src/GraphQL/Authorization/GraphQLAuthorizeAttribute.cs

```diff
@@ -19,18 +26,52 @@ public GraphQLAuthorizeAttribute(string policy)
         /// <summary>
         /// The name of policy to apply.
         /// </summary>
-        public string Policy { get; }
+        public string? Policy { get; set; }
+
+        /// <summary>
+        /// A comma-separated list of the roles to apply.
+        /// Role names will be trimmed before adding.
+        /// </summary>
+        public string? Roles { get; set; }
 
         /// <inheritdoc />
-        public override void Modify(TypeConfig type) => type.AuthorizeWith(Policy);
+        public override void Modify(TypeConfig type)
+        {
+            if (Policy != null)
+                type.AuthorizeWith(Policy);
```

Comment:
I would rename `AuthorizeWith` to `AuthorizeWithPolicy` to align with `AuthorizeWithRoles` name.

Reply:
That would be nice but it would be a breaking change.  I can add AuthorizeWithPolicy and optionally mark AuthorizeWith as obsolete, and probably update the v5 migration guide also since this change is so new.  Lmk.

Reply:
I know that it's breaking. Let's obsolete `AuthorizeWith`.

Reply:
Ok

Reply:
Well, all extension methods need to be obsoleted as well, I think.

Reply:
> Well, all extension methods need to be obsoleted as well, I think.

I'm not sure I follow.  I obsoleted all the AuthorizeWith methods.... didn't I?

Reply:
Yes, you did that.